indico/modules/events/templates/display/_common.html

```diff
@@ -0,0 +1,74 @@
+{% macro render_location(item, parent=none, class='') -%}
+    {% set show_venue = not parent or (parent.venue_name != item.venue_name) %}
+    {% set show_room = item.room_name != '' %}
+    {% set show_url = item.room and item.room.map_url %}
+    <span class="icon-location item-location {{ class }}">
+        {%- if show_url %}
+            <a href="{{ item.room.map_url }}">
+        {%- endif -%}
+        {%- if show_venue and show_room -%}
```

Comment:
For better readability I'd move this to a `render_location_text` macro and have:

```jinja
{% if show_url %}
    <a ...>{{ render_location_text(...) }}</a>
{% else %}
    {{ render_location_text(...) }}
{% endif %}
```